### PR TITLE
Yifanzhou/purview datamap 0901

### PR DESCRIPTION
**Pull request by yifan-zhou922**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/purview/azure-purview-datamap/README.md
Comment (msyyc) [inline sdk/purview/azure-purview-datamap/README.md:3]: Please complete README.md based on generated one.

File: sdk/purview/azure-purview-datamap/tests/conftest.py
Comment (msyyc) [inline sdk/purview/azure-purview-datamap/tests/conftest.py:38]: remove these lines since Py<3.8 is not supported anymore

File: sdk/purview/azure-purview-datamap/tests/test_smoke.py
Comment (msyyc) [inline sdk/purview/azure-purview-datamap/tests/test_smoke.py:20]: If you really run live test, there shall be recording files in the PR then please follow https://github.com/Azure/azure-sdk-for-python/blob/main/doc/dev/recording_migration_guide.md to upload recordings.

File: sdk/purview/azure-purview-datamap/tsp-location.yaml
Comment (msyyc) [inline sdk/purview/azure-purview-datamap/tsp-location.yaml:4]: Please follow https://github.com/Azure/azure-sdk-for-python/wiki/What-to-do-after-generating-the-SDK-code-with-codegen#createupdate-the-ciyml to update ci.yml

File: sdk/purview/azure-purview-datamap/tests/test_smoke.py
Comment (yifan-zhou922) [reply sdk/purview/azure-purview-datamap/tests/test_smoke.py:20]: Recording files updated

File: sdk/purview/azure-purview-datamap/README.md
Comment (yifan-zhou922) [reply sdk/purview/azure-purview-datamap/README.md:1]: updated

File: sdk/purview/azure-purview-datamap/tests/conftest.py
Comment (yifan-zhou922) [reply sdk/purview/azure-purview-datamap/tests/conftest.py]: updated

File: sdk/purview/azure-purview-datamap/tsp-location.yaml
Comment (yifan-zhou922) [reply sdk/purview/azure-purview-datamap/tsp-location.yaml:4]: updated

File: sdk/purview/azure-purview-datamap/azure/purview/__init__.py
Comment (msyyc) [inline sdk/purview/azure-purview-datamap/azure/purview/__init__.py:1]: Please try to fix CI failure with https://github.com/Azure/azure-sdk-for-python/wiki/How-to-fix-CI

File: sdk/purview/azure-purview-datamap/tests/test_smoke.py
Comment (msyyc) [reply sdk/purview/azure-purview-datamap/tests/test_smoke.py:20]: Which file is recording file in this PR?

Comment (yifan-zhou922) [reply sdk/purview/azure-purview-datamap/tests/test_smoke.py:20]: Please help with the error.
image

Comment (yifan-zhou922) [reply sdk/purview/azure-purview-datamap/tests/test_smoke.py:20]: I could only generate assets.json with empty tag.
Please help with this. 
image

File: sdk/purview/azure-purview-datamap/azure/purview/__init__.py
Comment (yifan-zhou922) [reply sdk/purview/azure-purview-datamap/azure/purview/__init__.py:1]: updated

File: tools/azure-sdk-tools/ci_tools/environment_exclusions.py
Comment (msyyc) [inline tools/azure-sdk-tools/ci_tools/environment_exclusions.py:36]: Please don't add your package in the list 
image

Comment (yifan-zhou922) [reply tools/azure-sdk-tools/ci_tools/environment_exclusions.py]: I am following this TSG to fix the Pylint/Mypy/Pyright failures. Please help suggest how should we resolve the failures. Thanks.
image

File: sdk/purview/azure-purview-datamap/README.md
Comment (msyyc) [reply sdk/purview/azure-purview-datamap/README.md:1]: [CI](https://dev.azure.com/azure-sdk/public/_build/results?buildId=3517353&view=logs&jobId=b70e5e73-bbb6-5567-0939-8415943fadb9&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=aea0e0e1-1e00-5c1b-bba8-b2b238976da3) shows README lacks some parts

File: tools/azure-sdk-tools/ci_tools/environment_exclusions.py
Comment (msyyc) [reply tools/azure-sdk-tools/ci_tools/environment_exclusions.py]: You could follow https://github.com/Azure/azure-sdk-for-python/blob/main/doc/eng_sys_checks.md#the-pyprojecttoml to fix it for now (e.g. https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/purview/azure-purview-sharing/pyproject.toml)

File: sdk/purview/azure-purview-datamap/samples/README.md
Comment (msyyc) [inline sdk/purview/azure-purview-datamap/samples/README.md:17]:
```suggestion
* Python 3.8 or later is required to use this package
```

File: sdk/purview/azure-purview-datamap/samples/README.md
Comment (msyyc) [inline sdk/purview/azure-purview-datamap/samples/README.md:39]: image
Where is the url?

File: sdk/purview/azure-purview-datamap/samples/get_types.py
Comment (catalinaperalta) [inline sdk/purview/azure-purview-datamap/samples/get_types.py:8]: You need to import sys to fix this error: https://dev.azure.com/azure-sdk/public/_build/results?buildId=3528665&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=e3f8f308-6e5f-58aa-3a2a-fe425958de32&l=369 

similarly the CI errors are pointing out that LOG on line 23 is not defined

File: sdk/purview/azure-purview-datamap/README.md
Comment (msyyc) [inline sdk/purview/azure-purview-datamap/README.md:64]: Where is the link of `request_builders_and_client`?

**Comment by msyyc** [comment]
@yifan-zhou922 Please resolve the conflict.

**Review by msyyc — APPROVED** [review]

File: sdk/purview/azure-purview-datamap/CHANGELOG.md
Comment (msyyc) [inline sdk/purview/azure-purview-datamap/CHANGELOG.md:3]: Please update date to latest before merge.

Comment (yifan-zhou922) [reply sdk/purview/azure-purview-datamap/CHANGELOG.md]: Thanks. Updated

**Comment by yifan-zhou922** [comment]
Please help merge the PR if there are no more comments.
cc @catalinaperalta / @iscai-msft

File: sdk/purview/azure-purview-datamap/README.md
Comment (catalinaperalta) [inline sdk/purview/azure-purview-datamap/README.md:6]:
```suggestion
### Installing the package
```

File: sdk/purview/azure-purview-datamap/README.md
Comment (catalinaperalta) [inline sdk/purview/azure-purview-datamap/README.md:2]: Nit I see the client name is DataMapClient also, do you need to call out `data-plane` in the description here? Are customers aware of this distinction?
```suggestion
Microsoft Purview Data Map provides the foundation for data discovery and data governance. Microsoft Purview Data Map is a cloud native PaaS service that captures metadata about enterprise data present in analytics and operation systems on-premises and cloud. DataMapClient provides a set of APIs in the Purview Data Map service. For a full list of APIs, please refer to [Data Map API](https://learn.microsoft.com/rest/api/purview/datamapdataplane/operation-groups?view=rest-purview-datamapdataplane-2023-09-01).
```

File: sdk/purview/azure-purview-datamap/pyproject.toml
Comment (catalinaperalta) [inline sdk/purview/azure-purview-datamap/pyproject.toml:2]: Why is this check turned off?

File: sdk/purview/azure-purview-datamap/samples/get_types.py
Comment (catalinaperalta) [inline sdk/purview/azure-purview-datamap/samples/get_types.py:10]: These environment variable names dont match the ones actually used in the sample

File: sdk/purview/azure-purview-datamap/samples/get_types.py
Comment (catalinaperalta) [inline sdk/purview/azure-purview-datamap/samples/get_types.py:16]: Do you need to set this?

File: sdk/purview/azure-purview-datamap/pyproject.toml
Comment (yifan-zhou922) [reply sdk/purview/azure-purview-datamap/pyproject.toml:2]: Suggested by @msyyc  to bypass a failure for pylink.

File: sdk/purview/azure-purview-datamap/samples/get_types.py
Comment (catalinaperalta) [reply sdk/purview/azure-purview-datamap/samples/get_types.py:16]: Customers can easily get this value?

Comment (catalinaperalta) [reply sdk/purview/azure-purview-datamap/samples/get_types.py:16]: Are customers mainly using the df environment or the prod env?

Comment (catalinaperalta) [reply sdk/purview/azure-purview-datamap/samples/get_types.py:16]: Wont block this PR on this question since you've tested this on your end, but if this is only necessary for specific scenarios I would recommend having a specific sample for those environments OR add a comment to explain when this environment variable should be used and to make this more easily consumable for customers you could add the default value for this environment variable for folks.

Comment (yifan-zhou922) [reply sdk/purview/azure-purview-datamap/samples/get_types.py:16]: Thanks for the suggestion. We will improve this as the follow-up items

**Review by catalinaperalta — APPROVED** [review]
Please note you should also have async tests and samples, you can add those in a separate PR though

**Pull request merged**
